core/src/main/java/org/elasticsearch/common/xcontent/support/AbstractXContentParser.java

```diff
@@ -77,23 +77,46 @@ public boolean isBooleanValue() throws IOException {
         switch (currentToken()) {
             case VALUE_BOOLEAN:
                 return true;
+            case VALUE_STRING:
+                return Booleans.isBoolean(textCharacters(), textOffset(), textLength());
+            default:
+                return false;
+        }
+    }
+
+    @Override
+    public boolean booleanValue() throws IOException {
+        Token token = currentToken();
+        if (token == Token.VALUE_STRING) {
+            return Booleans.parseBoolean(textCharacters(), textOffset(), textLength(), false /* irrelevant */);
+        }
+        return doBooleanValue();
+    }
+
+    @Override
+    @SuppressWarnings("deprecated")
```

Comment:
I'd probably mark this as `@Deprecated` rather than suppress if that is possible.